File: lib/models/client_model.dart
class ClientModel {
  final String id;
  final String name;
  final String email;
  final String phone;
  final String cnic;
  final String address;
  final String notes;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final DateTime? archivedAt;

  ClientModel({
    required this.id,
    required this.name,
    required this.email,
    required this.phone,
    required this.cnic,
    required this.address,
    required this.notes,
    required this.createdAt,
    this.updatedAt,
    this.archivedAt,
  });

  factory ClientModel.fromJson(Map<String, dynamic> json) {
    return ClientModel(
      id: json['id'] as String,
      name: json['name'] as String,
      email: json['email'],
      phone: json['phone'],
      cnic: json['cnic'],
      address: json['address'],
      notes: json['notes'],
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt:
          json['updatedAt'] != null ? DateTime.parse(json['updated_at']) : null,
      archivedAt: json['archived_at'] != null
          ? DateTime.parse(json['archived_at'])
          : null,
    );
  }
}

File: app/models/client_model.py
from sqlalchemy import Column, String, DateTime, func, Text, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.database.base import Base
import uuid

class Client(Base):
    __tablename__ = "client" 

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id= Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    name = Column(String, nullable=False, index=True)
    email = Column(String, index=True)  
    phone = Column(String, index=True)
    cnic = Column(String) 
    address = Column(String)
    notes = Column(Text)

    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    archived_at = Column(DateTime(timezone=True), nullable=True)

    
    cases = relationship(
        "CaseRelatedClient",
        back_populates="client",
    )
